src/sgame/sg_bot_util.cpp

```diff
@@ -504,7 +533,7 @@ AINodeStatus_t BotActionEvolve ( gentity_t *self, AIGenericNode_t* )
 
 	for ( auto const& cl : classes )
 	{
-		if ( cl.authorized.Get() && BotCanEvolveToClass( self, cl.item ) && BotEvolveToClass( self, cl.item ) )
+		if ( BotEvolveToClass( self, cl.item ) )
```

Comment:
The checks that the item is unlocked, enabled, and doesn't cost too much should not be removed.

Reply:
The evolution is done, in practice, by `BotEvolveToClass` and thus I think the checks belong there, instead of in the code deciding which class to select when none is specified.

Reply:
OK so you added enabled and unlocked checking there, but the price checks and also the prohibition on devolving was removed.

Reply:
This is done by `BotCanEvolveToClass` which is called by `BotEvolveToClass`.

In the original code, only the bot specific cvar was not checked by `BotEvolveToClass`, the `BotCanEvolveToClass` was (and still is) already called there.
I added the canBuyNow() (and a check that the class actually exists, I guess this was because it can be called from BT directly, I don't remember all details, that's old code to me) there, which only, in practice, adds the cvar check, since the unlocked check is one of the several checks done by BotCanEvolveToClass:

* unlocked (manual way)
* no devolve (which is probably because originally aliens could not devolve, no longer true, I'll have to implement this for fire extinguishing grangers, some day)
* enough credits

IMO this could use some cleanup too, ofc, but that's not the point of this PR (which is only to implement a way to allow admins to disable `bot repairs`).

Reply:
Oh I didn't see that BotCanEvolveToClass is called in that function. LGTM

Reply:
Understandable, the code repeats itself a lot.